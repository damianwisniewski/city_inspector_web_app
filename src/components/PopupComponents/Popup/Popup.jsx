import React from 'react'
import PropTypes from 'prop-types'

import './Popup.scss'
import CloseButton from '../CloseButton/CloseButton'
import Overlay from '../Overlay/Overlay'

const Popup = ({ darkOverlay, closeButton, onButtonClose, onOverlayClick, children }) => {
	return (
		<Overlay onOverlayClick={onOverlayClick} dark={darkOverlay || false}>
			<div className='popup-container'>
				{closeButton && <CloseButton onClose={onButtonClose} />}
				{children}
			</div>
		</Overlay>
	)
}

Popup.propTypes = {
	darkOverlay: PropTypes.bool,
	children: PropTypes.element.isRequired,
	closeButton: PropTypes.bool,
	onClose: PropTypes.func,
}

export default Popup
