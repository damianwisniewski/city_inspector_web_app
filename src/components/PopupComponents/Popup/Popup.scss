@import '../../../assets/styles/index';

.popup-container {
	display: flex;
	position: relative;
	min-width: 300px;
	min-height: 200px;
	justify-content: center;
	align-content: stretch;
	background-color: $white;
	border-radius: 5px;
	box-shadow: $shadow;
	padding: 15px;
	animation: 0.2s 1 forwards resize;

	& > *:not(button) {
		min-width: 100%;
		min-height: 100%;
		border-radius: 5px;
		overflow: hidden;
	}
}

@keyframes resize {
	from {
		transform: scale(0);
	}

	to {
		transform: scale(1);
	}
}
