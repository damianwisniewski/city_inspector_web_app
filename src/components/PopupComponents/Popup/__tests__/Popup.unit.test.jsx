import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Popup from '../Popup'

configure({ adapter: new Adapter() })

describe('<NavItem />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let PopupComponent
	const children = <div />

	beforeAll(() => {
		PopupComponent = shallow(<Popup>{children}</Popup>)
	})

	test('should render correctly without specyfi any props', () => {
		expect(toJson(PopupComponent)).toMatchSnapshot()
	})

	test('should pass props for overlay to Overlay component', () => {
		PopupComponent.setProps({
			darkOverlay: true,
			onOverlayClick: () => {},
		})

		expect(toJson(PopupComponent)).toMatchSnapshot()
	})

	test('should render also CloseButton for property closeButton=true', () => {
		PopupComponent.setProps({
			closeButton: true,
			onButtonClose: () => {},
		})

		expect(toJson(PopupComponent)).toMatchSnapshot()
	})
})
