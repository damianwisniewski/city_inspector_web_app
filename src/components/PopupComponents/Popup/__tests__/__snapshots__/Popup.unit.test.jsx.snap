// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should pass props for overlay to Overlay component 1`] = `
<Overlay
  dark={true}
  onOverlayClick={[Function]}
>
  <div
    className="popup-container"
  >
    <div />
  </div>
</Overlay>
`;

exports[`<NavItem /> should render also CloseButton for property closeButton=true 1`] = `
<Overlay
  dark={true}
  onOverlayClick={[Function]}
>
  <div
    className="popup-container"
  >
    <CloseButton
      onClose={[Function]}
    />
    <div />
  </div>
</Overlay>
`;

exports[`<NavItem /> should render correctly without specyfi any props 1`] = `
<Overlay
  dark={false}
>
  <div
    className="popup-container"
  >
    <div />
  </div>
</Overlay>
`;
